### PR TITLE
addition of parameter ‘unclipped’

### LINKED ISSUE #20: finalize and document "unclipped geometry mode"
In v1.0, there exists an (undocumented) feature flag in the data extraction endpoints, which disables the clipping to the query area-of-interest of the geojson feature geometries.

[The currently undocumented implementation of having it mixed up into the `properties` parameter is not great, because this parameter should just alter the `properties` of the returned GeoJSON, nothing more but also nothing less than that. `unclipped` doesn't fit into this parameter, because it alters the `geometry` of the resulting GeoJSON.]

In order to complete this feature, we need to:
 
* [ ] choose and create a dedicated parameter for it
* [ ] add it to the documentation

see also https://github.com/GIScience/ohsome-api/pull/18#issuecomment-663558742

### PR DESCRIPTION
**Pull request by FabiKo117**
implementing processing and adding it to the logic
adding it to the docs (swagger and sphinx)
refactoring boolean parameter usage
adding test
closes #20

### CODE REVIEW

File: docs/endpoints.rst [hunk 1]
```diff
@@ -1030,7 +1030,9 @@ Extraction Endpoints
 
    :query time: required; format same as described in time_
    :query properties: specifies what properties should be included for each feature representing an OSM element: ‘tags’ and/or 'metadata’; multiple values can be delimited by commas; default: empty
+   :query unclipped: boolean operator to apply a clipping to the geometries of the returned featues (‘true’), or not (‘false’); default: ‘false’
```
Comment (tyrasd) [inline docs/endpoints.rst:1033]: I'm wondering if this is clear enough… maybe we can tell what the geometries will be clipped against?

**Review by tyrasd — COMMENTED** [review]
looks good. small suggestion: maybe we should put a comment in the code that the old (undocumented) variant with `properties=unclipped` is deprecated, so we don't forget to remove it in the next release(s). perhaps it's possible to use `@Deprecated` for that?

File: docs/endpoints.rst (hunk 1 above)
Comment (FabiKo117) [reply docs/endpoints.rst:1033]: You mean something like ".. apply a clipping on the returned features using the defined spatial boundary..."?

**Comment by FabiKo117** [comment]
> looks good. small suggestion: maybe we should put a comment in the code that the old (undocumented) variant with `properties=unclipped` is deprecated, so we don't forget to remove it in the next release(s). perhaps it's possible to use `@Deprecated` for that?

I've tried this already, but can't find a proper position where to add the @Deprecated annotation. Can you find one maybe?

File: docs/endpoints.rst (hunk 1 above)
Comment (tyrasd) [reply docs/endpoints.rst:1033]: i would have maybe written something like: _“boolean operator to specify whether the returned features' geometries should be clipped to the query's spatial boundary (‘false’), or not (‘true’); default: ‘false’”_

Comment (tyrasd) [reply docs/endpoints.rst:1033]: PS: the values `true`/`false` are reversed in the description above: `true` would be for unclipped features, while `false` returns the clipped ones.

**Comment by tyrasd** [comment]
> > looks good. small suggestion: maybe we should put a comment in the code that the old (undocumented) variant with `properties=unclipped` is deprecated, so we don't forget to remove it in the next release(s). perhaps it's possible to use `@Deprecated` for that?
> 
> I've tried this already, but can't find a proper position where to add the @deprecated annotation. Can you find one maybe?
> Cause it's only relevant in line [464](https://github.com/GIScience/ohsome-api/blob/3628d042df46634d31e8f152354689ef1f1d5b08/src/main/java/org/heigit/ohsome/ohsomeapi/inputprocessing/InputProcessor.java#L464) in InputProcessor.java.

I believe it should be possible if one rewrites the whole `if - else if - else` statement slightly, like this:

```java
    for (String property : properties) {
      @Deprecated boolean oldUnclippedParameter = "unclipped".equalsIgnoreCase(property)
      if ("tags".equalsIgnoreCase(property)) {
        this.includeTags = true;
      } else if ("metadata".equalsIgnoreCase(property)) {
        this.includeOSMMetadata = true;
      } else if (oldUnclippedParameter) {
        this.unclipped = true;
      } else {
        throw new BadRequestException(ExceptionMessages.PROPERTIES_PARAM);
      }
    }
```

**Comment by tyrasd** [comment]
aah, regarding my comment above https://github.com/GIScience/ohsome-api/pull/29#discussion_r473817843, I realize now that it could result in double-negatives sometimes: `unclipped=false` could be a bit confusing. Maybe turn the whole thing around, and just call this parameter `clipGeometries` (or so)?!

**Comment by FabiKo117** [comment]
> aah, regarding my comment above [#29 (comment)](https://github.com/GIScience/ohsome-api/pull/29#discussion_r473817843), I realize now that it could result in double-negatives sometimes: `unclipped=false` could be a bit confusing. Maybe turn the whole thing around, and just call this parameter `clipGeometries` (or so)?!

that's a good point yes.. then it would be better to call it "clipGeometries"

**Pull request merged**